packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol

```diff
@@ -0,0 +1,361 @@
+
+        // Mock the superchain factory to return address(0)
+        _mockDeployments(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY, _to, address(0));
+
+        // Expect the revert with `InvalidSuperchainAddress` selector
+        vm.expectRevert(InvalidSuperchainAddress.selector);
+
+        // Act
+        l2StandardBridge.convert(_from, _to, _amount);
+    }
+
+    /// @notice Test that the `convert` function with different remote tokens reverts
+    function testFuzz_convert_differentRemoteAddresses_reverts(
+        address _from,
+        address _to,
+        uint256 _amount,
+        address _fromRemoteToken,
+        address _toRemoteToken
+    )
+        public
+    {
+        // Assume
+        vm.assume(_fromRemoteToken != address(0));
+        vm.assume(_toRemoteToken != address(0));
+        vm.assume(_fromRemoteToken != _toRemoteToken);
+
+        // Arrange
+        _setUpLegacyToSuper(_from, _to);
+
+        // Mock the legacy factory to return `_fromRemoteToken`
+        _mockDeployments(address(l2OptimismMintableERC20Factory), _from, _fromRemoteToken);
+
+        // Mock the superchain factory to return `_toRemoteToken`
+        _mockDeployments(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY, _to, _toRemoteToken);
+
+        // Expect the revert with `InvalidTokenPair` selector
+        vm.expectRevert(InvalidTokenPair.selector);
+
+        // Act
+        l2StandardBridge.convert(_from, _to, _amount);
+    }
+
+    /// @notice Test that the `convert` function succeeds
+    function testFuzz_convert_succeeds(
+        address _caller,
+        address _from,
+        address _to,
+        uint256 _amount,
+        address _remoteToken
+    )
+        public
+    {
+        // Assume
+        vm.assume(_remoteToken != address(0));
+
+        // Arrange
+        _setUpLegacyToSuper(_from, _to);
+
+        // Mock the legacy and superchain factory to return `_remoteToken`
+        _mockDeployments(address(l2OptimismMintableERC20Factory), _from, _remoteToken);
+        _mockDeployments(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY, _to, _remoteToken);
+
+        // Mock the `burn` and `mint` functions
+        vm.mockCall(_from, abi.encodeWithSelector(MintableAndBurnable.burn.selector, _caller, _amount), abi.encode());
+        vm.mockCall(_to, abi.encodeWithSelector(MintableAndBurnable.mint.selector, _caller, _amount), abi.encode());
+
+        // Expect the `Converted` event to be emitted
+        vm.expectEmit(true, true, true, true, address(l2StandardBridge));
+        emit Converted(_from, _to, _caller, _amount);
+
+        // Expect the `mint` and `burn` functions to be called
+        vm.expectCall(_from, abi.encodeWithSelector(MintableAndBurnable.burn.selector, _caller, _amount), 1);
+        vm.expectCall(_to, abi.encodeWithSelector(MintableAndBurnable.mint.selector, _caller, _amount), 1);
+
+        // Act
+        vm.prank(_caller);
+        l2StandardBridge.convert(_from, _to, _amount);
+    }
```

Comment:
On other tests (e.g. `OptimismSuperchainERC20.t.sol`), we are asserting balances after mints and burns, and they're missing here

Reply:
Hmm this has to be tested in a kind of integration/e2e (or be added later) because in this branch we dont have neither the SuperchainERC20Factory nor the SuperchainERC20, that's why we are mocking the `burn` and `mint` functions

Reply:
yeah, makes sense